engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCachingSlickDatabaseSpec.scala

```diff
@@ -32,7 +33,13 @@ class CallCachingSlickDatabaseSpec
   DatabaseSystem.All foreach { databaseSystem =>
     behavior of s"CallCachingSlickDatabase on ${databaseSystem.name}"
 
-    lazy val dataAccess = DatabaseTestKit.initializedDatabaseFromSystem(EngineDatabaseType, databaseSystem)
+    val containerOpt: Option[Container] = DatabaseTestKit.getDatabaseTestContainer(databaseSystem)
+
+    lazy val dataAccess = DatabaseTestKit.initializeDatabaseByContainerOptTypeAndSystem(containerOpt, EngineDatabaseType, databaseSystem)
+
+    it should "start container if required" taggedAs DbmsTest in {
+      containerOpt.foreach { _.start }
+    }
```

Comment:
TOL would these `start...` and `stop..` methods make sense as `beforeAll` and `afterAll` methods of the `BeforeAndAfterAll` trait?

Reply:
That would be nice, but the problem is that we need to run the same set of tests described in a single `*Spec` against a list of databases and that would imply a single `beforeAll/afterAll` call for the entire spec. However we need to start/stop the corresponding container for each database, i.e. multiple times within the `*Spec`.